docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,69 @@
+---
+id: best-practices
+title: Best practices
+description: An index of best practices for plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - best practices
+sidebar_position: 0.5
+---
+
+# Best practices for plugin development
+
+We have compiled a list of good practices for building high-quality plugins for Grafana. We recommend referring to them when building your own plugin for best results.
+
+Is something missing from this list? [Let us know](https://github.com/grafana/plugin-tools/issues/new).
+
+## General
+
+- **Verify that your data source or app plugin can be provisioned.** Refer to [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- **Include default dashboards in your data source or app plugin.** Refer to [Bundling of dashboards](https://grafana.com/developers/plugin-tools/introduction/plugin-types-usage#bundling-of-dashboards) for more information.
+- **Ensure that the minimum version of Grafana is correct.** Make sure that the `grafanaDependency` in your `plugin.json` points to the earliest version of Grafana that your plugin fully supports.
+- **Consider building an [app plugin](/introduction/plugin-types-usage#app-plugins).** App plugins offer more flexibility and additional capabilities than panel or data source plugins, allowing you to build more tailored monitoring experiences for a given service. Apps provide a better user experience than simply including default dashboards with a data source.
```

Comment:
>When reading this I get the impression that I should always build an app rather than panel or datasource. Or is that we (you) want?

That's valid feedback, it's perhaps too strongly worded. The intent was just to highlight that apps do offer a lot more in terms of capabilities and to encourage developers, particularly say ISVs, that if they take advantage of those capabilities then they can make it significantly easier for users to interact with their service.

🤔 of a rewrite

Reply:
wdyt about:


```
**Consider building an [app plugin](/introduction/plugin-types-usage#app-plugins).**

If your users would benefit from more of an opinionated experience then an app plugin may be an appropriate choice, offering both increased flexibility and additional capabilities like custom pages and UI extensions.
```

Reply:
If this doesn't solve the concern, remove this bullet to unblock the PR

Reply:
"opinionated experience" seems a bit vage for me. Since it's not a very specific best-practice lets just remove it. This info should be already in the intro page for plugin types.

Reply:
Go for it